requirements.txt

```diff
@@ -1,36 +1,3 @@
-tfx==1.10.0
-tensorflow-gnn==0.2.1
```

Comment:
You shouldn't take out all these other requirements since they are needed for downstream projects. E.g. there are models that  require tensorflow-gnn and it's mendokusai to have to install these every time you rebuild the image in the project directory.

Reply:
it won't be build every time
i started to use techniques for docker caching
docker doesn't run a docker layer if there are no changes of file itself
it will cache all the layers except if the requirements.txt is changed
in that case it will re build with installing those requirements again

Reply:
yeah but because we do pip install ., anytime there is a change to the any of the project's files (model.py, preprocess.py, etc.) that layer has to be rebuilt so it isn't cached. all the requirements need to be reinstalled. It takes quite a long time for some of these libraries. This is really the main reason to splitting this out as a base image, so that those time-consuming build layers are frozen.

Reply:
in the main library
i took out installing requirements from the setup.py
i knew we had this issue
so in the dockerfile
the requirements will be installed first and then the main repo
this will enable the cache of the dockerfile and save build time

Reply:
i will make the PR there so you can have better picture

Reply:
I see. Ok that should work then.